### PR TITLE
expression: fix datarace for `builtinIlikeSig`

### LINKED ISSUE #49677: ilike function may cause data race for non-constant pattern
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

We can see that in line 201, the pattern is constructed and set `builtinIlikeSig.pattern`

https://github.com/pingcap/tidb/blob/5c279d800774ae7ce5ff9934e1334740f1c7207b/pkg/expression/builtin_ilike_vec.go#L200-L203

`contVec` and `vecVec` uses `builtinlikeSIg.pattern` directly:

https://github.com/pingcap/tidb/blob/5c279d800774ae7ce5ff9934e1334740f1c7207b/pkg/expression/builtin_ilike_vec.go#L137-L150

If expression is evaluated in concurrency, it may cause data race

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #49677

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
fix datarace for `ilike` function in same cases
```

### CODE REVIEW

**Review by YangKeao — APPROVED** [review]
LGTM

File: pkg/expression/builtin_ilike_vec.go
Comment (xzhangxian1008) [inline pkg/expression/builtin_ilike_vec.go:1]: Maybe we need to add some concurrent tests.

**Review by xzhangxian1008 — APPROVED** [review]
Other LGTM

**Review by windtalker — APPROVED** [review]
LGTM

**Pull request merged**